allennlp/data/dataset_readers/dataset_reader.py

```diff
@@ -271,17 +295,50 @@ def deserialize_instance(self, string: str) -> Instance:
         """
         return jsonpickle.loads(string)  # type: ignore
 
-    def ensure_instance(self, raw_instance) -> Instance:
-        if isinstance(raw_instance, Instance):
-            return raw_instance
-        return self.text_to_instance(**raw_instance)
+    def parse_raw_data(self, raw_data: RawData) -> Instance:
```

Comment:
Was this from someone else's feedback? What's the benefit of having additional raw data types here? I feel like `Dict[str, Any] || Instance` is ok.

Reply:
The idea being that `_read` should do as little pre-processing as possible. See https://github.com/allenai/allennlp/blob/multi-process-loading/allennlp/data/dataset_readers/sequence_tagging.py#L71 for example.

Reply:
Hmm, I think this is too extreme. You can still do this with a single Dict type in a _new_ dataset reader by doing `{"data": blob}` and having your `text_to_instance` method accept this. It isn't worth the API complexity of handling the different types and the errors that users will encounter from generically typed dataset readers.

Reply:
Hm alright, well I think it's worth supporting a `Tuple[Any, ...]` as well, which would be treated as positional arguments to `text_to_instance`. It's less verbose to generate tuples as opposed to dictionaries, and probably a little more efficient.

Reply:
I'd recommend you remove that for simplicities sake, as well as removing the ability for users to shoot themselves in the foot in the case that they re-order the arguments to `text_to_instance`. It's very likely that some args are compatible with each other, and this would be a really frustrating bug to run into, just because the argument order needs to be defined in two places in the `DatasetReader`.